File: Builder/Golang/Report.go
package main

import "fmt"

type Report struct {
	filename string
	content string
}

func (report *Report) setFilename(filename string)  {
	report.filename = filename
}

func (report *Report) setContent(content string)  {
	report.content = content
}

func (report *Report) describe() {
	if report.filename != "" {
		fmt.Println("The name of the file is", report.filename)
	} else {
		fmt.Println("The name file doesn't have any name")
	}

	if report.content != "" {
		fmt.Println("The content of the file is", report.content)
	} else {
		fmt.Println("The name file doesn't have any content")
	}
}

File: Builder/Javascript/Report.js
class Report {
  describe() {
    if (this.filename) {
      console.log("The name of the file is " + this.filename);
    } else {
      console.log("The file doesn't have a name");
    }

    if (this.content) {
      console.log("The content of the file is " + this.content);
    } else {
      console.log("The file doesn't have a content");
    }
  }
}

export default Report;